components/Favorites/useToggleFav.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect } from 'react';
+
+
+
+/*
+export default function FavStatus( props ) {
+    
+    const [ isFav, toggleFav ] = useToggleMovieFav( props.id );
+
+}
+*/
+
+export default function useToggleFav ( key: string, id: number ) {
+
+    const [ isFav, setIsFav ] = useState( () => {
+
+        if ( typeof window !== "undefined" ) {
```

Comment:
Make sure you return a value if there is no window. see https://usehooks.com/useLocalStorage/
In your case just false.

Try to avoid nested if else...

Reply:
Fixed this on commit ecb83b4841e350f06c8ec342069ca6ad62ef7044